src/CollisionHandler.cpp

```diff
@@ -1,76 +1,109 @@
+            return returnObjects.at(x);
         }
-      }
+    }
     return nullptr;
 }
 
 // Check for collisions during movement
-bool CollisionHandler::detectMovementCollision(HitBox* hb) {
-    std::vector<HitBox*> returnObjects;
-    returnObjects = quadtreeMov->retrieve(returnObjects, hb);
-      for (unsigned int x = 0; x < returnObjects.size(); x++) {
-        if (returnObjects.at(x) != nullptr && hb->attached != returnObjects.at(x)->attached) {
-            if (SDL_HasIntersection(&hb->getRect(), &returnObjects.at(x)->getRect()) &&
-                !(hb->isPlayerFriendly() && returnObjects.at(x)->isPlayerFriendly()) ) {
-                return true;
-            }
+bool CollisionHandler::detectMovementCollision(const HitBox *hb) {
+    std::vector<HitBox *> returnObjects;
+    returnObjects = quadtreeMov.retrieve(hb);
+    for (unsigned int x = 0; x < returnObjects.size(); x++) {
+        if (returnObjects.at(x) != nullptr && hb->attached != returnObjects.at(x)->attached 
+            && SDL_HasIntersection(&hb->getRect(), &returnObjects.at(x)->getRect()) 
+                && !(hb->isPlayerFriendly() && returnObjects.at(x)->isPlayerFriendly()) ) {
+            return true;
         }
-      }
+    }
     return false;
 }
 
 //check for pickup collision
-HitBox* CollisionHandler::detectPickUpCollision(HitBox* hb) {
+HitBox *CollisionHandler::detectPickUpCollision(const HitBox *hb) {
     std::vector<HitBox*> returnObjects;
-    returnObjects = quadtreePickUp->retrieve(returnObjects, hb);
-      for (unsigned int x = 0; x < returnObjects.size(); x++) {
-        if (returnObjects.at(x) != nullptr && hb->attached != returnObjects.at(x)->attached) {
-            if (SDL_HasIntersection(&hb->getRect(), &returnObjects.at(x)->getRect()) &&
-                !(hb->isPlayerFriendly() && returnObjects.at(x)->isPlayerFriendly()) ) {
-                return returnObjects.at(x);
-            }
+    returnObjects = quadtreeMov.retrieve(hb);
+    for (unsigned int x = 0; x < returnObjects.size(); x++) {
+        if (returnObjects.at(x) != nullptr && hb->attached != returnObjects.at(x)->attached 
+            && SDL_HasIntersection(&hb->getRect(), &returnObjects.at(x)->getRect()) 
+                && !(hb->isPlayerFriendly() && returnObjects.at(x)->isPlayerFriendly()) ) {
+            return returnObjects.at(x);
         }
-      }
+    }
     return nullptr;
 }
 
+
+// Created by DericM 3/8/2017
+std::priority_queue<HitBox*> CollisionHandler::detectLineCollision(
+        Marine &marine, const int range) {
+    
+    const double degrees = marine.getAngle() - 90;
+    const double radians = degrees * M_PI / 180;
+    const int playerX = marine.getX() + (MARINE_WIDTH / 2);
+    const int playerY = marine.getY() + (MARINE_HEIGHT / 2);
+    const int deltaX  = range * cos(radians);
+    const int deltaY  = range * sin(radians);
+    int aX, aY, bX, bY;
+
+    std::vector<HitBox*> allHitBoxes;
```

Comment:
Can be condensed to:
' std::vector<HitBox *> allHitBoxes = quadtreePro.objects; '